src/types.ts

```diff
@@ -667,6 +667,41 @@ export type BanUserOptions<UserType = UnknownType> = UnBanUserOptions & {
   user_id?: string;
 };
 
+export type QueryBannedUsersResponse = APIResponse & {
+  bans?: BanResponse[];
+};
+
+export type BanResponse<
+  ChannelType = UnknownType,
+  CommandType extends string = LiteralStringForUnion,
+  UserType = UnknownType
+> = APIResponse & {
+  user: UserResponse<UserType>;
+  banned_by?: UserResponse<UserType>;
+  channel?: ChannelResponse<ChannelType, CommandType, UserType>;
+  expires?: string | Date;
+  ip_ban?: boolean;
+  reason?: string;
+  timeout?: number;
+};
+
+export type BannedFilters = {
+  banned_by_id?: string;
+  channel_cid?: string;
+  created_at?: string;
```

Comment:
you can filter by just a singular time?

Reply:
So these are different than the other filter types, just one object with a bunch of stuff?

Reply:
you can combine multiple filters, I need your help with the operator restrictions black magic

Reply:
`channel_cid`=> eq,In
`reason`=>  In,Nin,Eq,Ne,Gt,Gte,Lt,Lte,Exists,Autocomplete
`banned_by_id `=> In,Nin,Eq,Ne,Gt,Gte,Lt,Lte,Exists
`created_at `=> In,Nin,Eq,Ne,Gt,Gte,Lt,Lte,Exists
`user_id `=> In,Nin,Eq,Ne,Gt,Gte,Lt,Lte,Exists